### PR TITLE
Support Basic authentication

### PR DESCRIPTION
**Pull request by kamatama41**
Hi, thank you for sharing the library.
I'd like to use Hatena Blog API via Basic authentication.

ref http://developer.hatena.ne.jp/ja/documents/blog/apis/atom
> Basic認証についてはユーザ名としてはてなIDを、パスワードとしてAPIキーを利用することで認証できます。Basic認証はAPIのURLがhttpsではじまる場合にのみ利用できます。

To do so, I made web requesting encapsulated as `requester` and implemented a requester for Basic auth. Could you review it? Thank you.

### CODE REVIEW

File: lib/hatenablog/requester.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    class Basic
+      METHODS = {
+        get: Net::HTTP::Get,
+        post: Net::HTTP::Post,
+        put: Net::HTTP::Put,
+        delete: Net::HTTP::Delete
+      }
+
+      # Create a new OAuth 1.0a access token.
+      # @param [OAuth::AccessToken] access_token access token object
```
Comment (kymmt90) [inline lib/hatenablog/requester.rb]: - `Create a new OAuth 1.0a access token.`
  - Fix: `Create a new Basic authentication requester.`
- `@params [OAuth::AccessToken] access_token access token object`
  - Fix as follows:

```
@params [string] user_id Hatena user ID
@params [string] api_key Hatena API key
```

**Comment by kymmt90** [comment]
Thank you for your patch!
It looks good to me.
I will merge it soon:+1:

**Comment by kymmt90** [comment]
I have released v0.4.0 including this feature!

**Comment by kamatama41** [comment]
Thank you for releasing quickly 😃

**Pull request merged**